src/scipp/curve_fit.py

```diff
@@ -516,37 +615,65 @@ def curve_fit(
         for arg, coord in coords.items()
     }
 
-    p0 = _make_defaults(f, coords.keys(), p0)
-
-    f = (
-        _wrap_scipp_func(f, p0)
-        if not unsafe_numpy_f
-        else _wrap_numpy_func(f, p0, coords.keys())
-    )
-
     map_over = tuple(
         d
         for d in da.dims
         if d not in reduce_dims and not any(d in c.dims for c in coords.values())
     )
 
-    # Create a dataarray with only the participating coords
-    _da = DataArray(da.data, coords=coords, masks=da.masks)
-
-    out = _prepare_numpy_outputs(da, p0, map_over)
+    p0 = _make_defaults(f, coords.keys(), p0)
+    bounds = _parse_bounds(bounds, p0)
 
-    _curve_fit(
-        f,
-        _da,
-        p0,
-        _parse_bounds(bounds, p0),
-        map_over,
-        unsafe_numpy_f,
-        out,
-        **kwargs,
+    workers = (
+        # process_cpu_count added in 3.13
+        getattr(os, 'process_cpu_count', os.cpu_count)() if workers is None else workers
     )
 
-    return _datagroup_outputs(da, p0, map_over, *out)
+    pardim = None
+    if workers > 1 and len(map_over) > 0:
+        max_size = max((da.sizes[dim] for dim in map_over))
+        max_size_dim = next((dim for dim in map_over if da.sizes[dim] == max_size))
+        pardim = max_size_dim if max_size > 1 else None
```

Comment:
Here the dimension to slice into chunks and parallelize over is decided.
The dimension with the largest size is used because that gives us a more even load on the separate processes.

Reply:
We might want to add some kind of heuristic for when to not use multiprocessing and only rely on the built-in thread parallelism of scipp.

A heuristic that seems reasonable from experiments (on my laptop) is when the size of each fit becomes larger than 5000 points per core.

Reply:
Update: a heuristic has been added